src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -457,4 +475,28 @@ object Task {
       else None
     }
   }
+
+  implicit class TaskStatusComparison(val task: Task) extends AnyVal {
+    def isReserved: Boolean = compareTo(MarathonTaskStatus.Reserved)
+    def isCreated: Boolean = compareTo(MarathonTaskStatus.Created)
+    def isError: Boolean = compareTo(MarathonTaskStatus.Error)
+    def isFailed: Boolean = compareTo(MarathonTaskStatus.Failed)
+    def isFinished: Boolean = compareTo(MarathonTaskStatus.Finished)
+    def isKilled: Boolean = compareTo(MarathonTaskStatus.Killed)
+    def isKilling: Boolean = compareTo(MarathonTaskStatus.Killing)
+    def isLost: Boolean = compareTo(MarathonTaskStatus.Lost)
+    def isMesosLost: Boolean = isLost || isGone || isUnreachable
+    def isRunning: Boolean = compareTo(MarathonTaskStatus.Running)
+    def isStaging: Boolean = compareTo(MarathonTaskStatus.Staging)
+    def isStarting: Boolean = compareTo(MarathonTaskStatus.Starting)
+    def isUnreachable: Boolean = compareTo(MarathonTaskStatus.Unreachable)
+    def isGone: Boolean = compareTo(MarathonTaskStatus.Gone)
+    def isUnknown: Boolean = compareTo(MarathonTaskStatus.Unknown)
+
+    def mightBeLost: Boolean = MarathonTaskStatus.mightBeLost.contains(task.status.taskStatus)
```

Comment:
same thing as above: might be lost doesn't help us – for a transitional phase this might have value but it should be removed before merging.